### PR TITLE
Anonymize whole URLs

### LINKED ISSUE #14: Anonymize whole URLs
If a link is to a URL that partially matches a pattern I want to hide, the whole URL should probably be anonymized. Right now, I get `engineering.xxx.edu`, which causes the following issues:
* xxx might be a registered domain (`xxx.edu` doesn't exist, but it's the case for `.com` and a few other)
* those domains **are usually pornographic** (please don't make my repository link to pornographic material :wink:)
* the remaining parts of the URL are still there, and can be used to de-anonymize (not all companies have an `.edu` domain, or an `engineering` subdomain)

The link is broken anyway, better to anonymize completely. Also, it could point to a page/popup saying it's been anonymized, instead of leading to an error page, so the reader knows what's going on.

### PR DESCRIPTION
**Pull request by remram44**
I am unable to test this because of errors 500 (from github lib).

Fix #14

### CODE REVIEW

**Comment by tdurieux** [comment]
Thanks for the PR, I will try it locally

File: server.py [hunk 1]
```diff
@@ -108,9 +108,20 @@ def remove_terms(content, repository_configuration, word_boundaries=True):
...
-                    content = re.compile(term, re.IGNORECASE).sub("XXX", content)
+                    regex = re.compile(term, re.IGNORECASE)
+
+                if whole_urls:
+                    def sub_url(m):
+                        if regex.search(m.group(0)):
+                            return 'XXX'
+                        return m.group(0)
+
+                    url_regex = re.compile('\b((https?|ftp|file)://)[-A-Za-z0-9+&@#/%?=~_|!:,.;]+[-A-Za-z0-9+&@#/%=~_|]\b')
```
Comment (tdurieux) [inline server.py]:
```suggestion
                    url_regex = re.compile('((https?|ftp|file)://)[-A-Za-z0-9+&@#/%?=~_|!:,.;]+[-A-Za-z0-9+&@#/%=~_|]')
```

By changing this line, your code is able to anonymize the complete url

Comment (remram44) [reply server.py]: Maybe it should be `\\b`?

Comment (tdurieux) [reply server.py]: Indeed 👍

**Comment by tdurieux** [comment]
thanks a lot

**Pull request merged**